### PR TITLE
Fix GeoscalingWebClient create primary domain REST call

### PR DESCRIPTION
**Pull request by mikezaccardo**

### CODE REVIEW

**Comment by mikezaccardo** [comment]
The subdomain component of the tests work so they both pass now -- thanks @alasdairhodge

File: software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java [hunk 1]
```diff
@@ -42,8 +42,8 @@
     
     private final static String USERNAME = "cloudsoft";
     private final static String PASSWORD = "cl0uds0ft";
-    private final static String PRIMARY_DOMAIN = "domain"+((int)(Math.random()*10000))+".test.org";
-    private final static String SUBDOMAIN = "subdomain"+((int)(Math.random()*10000));
+    private final static String PRIMARY_DOMAIN = "domain"+((int)(Math.random()*10000) + 10000)+".test.org";
+    private final static String SUBDOMAIN = "subdomain"+((int)(Math.random()*10000) + 10000);
```
Comment (alasdairhodge) [inline software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: Why the constant offset? Trying to keep the lengths consistent?

**Comment by alasdairhodge** [comment]
Nice catch, Mike; thanks for digging into this :o)

File: software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java (hunk 1 above)
Comment (mikezaccardo) [inline software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: To guarantee that they don't collide with the existing domains (as unlikely as it is).

Comment (alasdairhodge) [inline software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: Hmm, I don't think that's necessary: the existing (what, four?) `*.test.org` domains look like remnants of previous failed test runs and should be deleted.

Comment (alasdairhodge) [inline software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: Actually, that makes me think we should review the domain cleanup logic in the test; looks like tear-down code wasn't executed following previous test failures...

File: software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java [hunk 2]
```diff
@@ -42,8 +42,8 @@
     
     private final static String USERNAME = "cloudsoft";
     private final static String PASSWORD = "cl0uds0ft";
-    private final static String PRIMARY_DOMAIN = "domain"+((int)(Math.random()*10000))+".test.org";
-    private final static String SUBDOMAIN = "subdomain"+((int)(Math.random()*10000));
+    private final static String PRIMARY_DOMAIN = "domain"+((int)(Math.random()*10000) + 10000)+".test.org";
```
Comment (ahgittin) [inline software/webapp/src/test/java/brooklyn/entity/dns/geoscaling/GeoscalingWebClientTest.java]: use `Strings.makeRandomId()` or `Strings.makeZeroPaddedString()` instead ?

**Comment by mikezaccardo** [comment]
Will improve cleanup logic, use better name generation, and remove vestigial domains from GeoScaling.

**Comment by mikezaccardo** [comment]
^ Done x3

**Comment by aledsage** [comment]
Thanks @mikezaccardo - looks good; merging.

**Pull request merged**